config/config_test.go

```diff
@@ -457,6 +488,48 @@ func TestTreeRootFile(t *testing.T) {
 	checkValue(tempDir, ".git/config")
 }
 
+func TestTreeRootCmd(t *testing.T) {
+	as := require.New(t)
+
+	cfg := &config.Config{}
+	v, flags := newViper(t)
+
+	checkValue := func(treeRoot string) {
+		readValue(t, v, cfg, func(cfg *config.Config) {
+			as.Equal(treeRoot, cfg.TreeRoot)
+		})
+	}
+
+	tempDir := t.TempDir()
+	as.NoError(os.MkdirAll(filepath.Join(tempDir, "foo"), 0o755))
+	as.NoError(os.MkdirAll(filepath.Join(tempDir, "bar"), 0o755))
+
+	// default with no flag, env or config
+	// should match the absolute path of the directory in which the config file is located
+	checkValue(filepath.Dir(v.ConfigFileUsed()))
```

Comment:
Should there also be a variant of this test-case for when `tempDir` is a git repo?

In that scenario we'd expect `checkValue(tempDir)`, right?

Reply:
I guess its fine to omit that test if this is intended to be more of a unit test, and there's integration tests elsewhere that cover the _"find tree root without any `tree-root-*` setting defined, while in a git repo"_ scenario?

I'm not too familiar with your test suite, but it'd be nice if that scenario had _some_ coverage somewhere :slightly_smiling_face: